toolchain/sem_ir/stringify_type.cpp

```diff
@@ -116,6 +106,33 @@ class StepStack {
     PushInstId(sem_ir_->types().GetInstId(type_id));
   }
 
+  // Pushes a sequence of items onto the stack. This handles reversal, such that
+  // the caller can pass items in print order instead of stack order.
+  //
+  // Note that with `ListSeparator`, the object's reference isn't stored, but
+  // the separator `StringRef` will be. That should be a constant though, so is
+  // safe.
+  auto Push(llvm::ArrayRef<PushItem> items) -> void {
+    for (auto item : llvm::reverse(items)) {
+      VariantMatch(
+          item, [&](InstId inst_id) { PushInstId(inst_id); },
+          [&](llvm::StringRef string) { PushString(string); },
+          [&](NameId name_id) { PushNameId(name_id); },
+          [&](ElementIndex element_index) { PushElementIndex(element_index); },
+          [&](QualifiedNameItem qualified_name) {
+            PushQualifiedName(qualified_name.first, qualified_name.second);
+          },
+          [&](EntityNameItem entity_name) {
+            PushEntityName(entity_name.first, entity_name.second);
+          },
+          [&](EntityNameId entity_name_id) {
+            PushEntityNameId(entity_name_id);
+          },
+          [&](TypeId type_id) { PushTypeId(type_id); },
+          [&](llvm::ListSeparator* sep) { PushString(*sep); });
+    }
+  }
```

Comment:
What do you think of adding something like:
```suggestion
  auto PushArray(llvm::ArrayRef<PushItem> items) -> void {
    for (auto item : llvm::reverse(items)) {
      VariantMatch(
          item, [&](InstId inst_id) { PushInstId(inst_id); },
          [&](llvm::StringRef string) { PushString(string); },
          [&](NameId name_id) { PushNameId(name_id); },
          [&](ElementIndex element_index) { PushElementIndex(element_index); },
          [&](QualifiedNameItem qualified_name) {
            PushQualifiedName(qualified_name.first, qualified_name.second);
          },
          [&](EntityNameItem entity_name) {
            PushEntityName(entity_name.first, entity_name.second);
          },
          [&](EntityNameId entity_name_id) {
            PushEntityNameId(entity_name_id);
          },
          [&](TypeId type_id) { PushTypeId(type_id); },
          [&](llvm::ListSeparator* sep) { PushString(*sep); });
    }
  }
  template<typename ...T>
  auto Push(T ...items) -> void {
    PushArray({items...});
  }
```
so that the calls to `Push` don't need braces?

Reply:
I'm hesitant about this, but can apply here and separately propose the API change that is the source of my hesitance.